master/internal/workspace/postgres_workspace.go

```diff
@@ -164,3 +180,62 @@ func GetAllNamespacesForRM(ctx context.Context, rmName string) ([]string, error)
 	}
 	return ns, nil
 }
+
+// AddWorkspaceNamespaceBinding adds a workspace-namespace binding.
+func AddWorkspaceNamespaceBinding(ctx context.Context, wkspNmsp *model.WorkspaceNamespace,
+	tx *bun.Tx,
+) error {
+	if tx != nil {
+		_, err := tx.NewInsert().Model(wkspNmsp).Exec(ctx)
+		if err != nil {
+			return fmt.Errorf("error adding workspace-namespace binding to database: %w", err)
+		}
+	} else {
+		_, err := db.Bun().NewInsert().Model(wkspNmsp).Exec(ctx)
+		if err != nil {
+			return fmt.Errorf("error adding workspace-namespace binding to database: %w", err)
+		}
+	}
+	return nil
+}
+
+// GetWorkspaceNamespaceBindings gets the workspace-namespace bindings for a given workspace.
+func GetWorkspaceNamespaceBindings(ctx context.Context,
+	wkspID int,
+) ([]model.WorkspaceNamespace, error) {
+	var workspaceNamespaceBindings []model.WorkspaceNamespace
+	err := db.Bun().NewSelect().
+		Model(&model.WorkspaceNamespace{}).
+		Where("workspace_id = ?", wkspID).
+		Scan(ctx, &workspaceNamespaceBindings)
+	if err != nil {
+		return nil, err
+	}
+	return workspaceNamespaceBindings, nil
+}
+
+// DeleteWorkspaceNamespaceBindings deletes the workspace-namespace binding.
+func DeleteWorkspaceNamespaceBindings(ctx context.Context, wkspID int,
+	clusterNames []string, tx *bun.Tx,
+) error {
+	if tx != nil {
+		_, err := tx.NewDelete().Model(&model.WorkspaceNamespace{}).
+			Where("workspace_id = ?", wkspID).
+			Where("cluster_name in (?)", bun.In(clusterNames)).
+			Exec(ctx)
+		if err != nil {
+			return fmt.Errorf(`error deleting workspace-namespace binding with workspace-id %d`,
+				wkspID)
+		}
+		return nil
+	}
+
+	_, err := db.Bun().NewDelete().Model(&model.WorkspaceNamespace{}).
+		Where("workspace_id = ?", wkspID).
+		Where("cluster_name in (?)", bun.In(clusterNames)).
+		Exec(ctx)
+	if err != nil {
+		return fmt.Errorf(`error deleting workspace-namespace binding with workspace-id %d`, wkspID)
+	}
+	return nil
+}
```

Comment:
```suggestion
func DeleteWorkspaceNamespaceBindingsTx(ctx context.Context, wkspID int,
	clusterNames []string, tx *bun.Tx,
) ([]model.WorkspaceNamespace, error) {
	var deletedBindings []model.WorkspaceNamespace
	_, err := tx.NewDelete().
		Model(&model.WorkspaceNamespace{}).
		Where("workspace_id = ?", wkspID).
		Where("cluster_name in (?)", bun.In(clusterNames)).
		Returning("*").
		Exec(ctx, &deletedBindings)
	if err != nil {
		return nil, fmt.Errorf(`error deleting workspace-namespace binding with workspace-id %d`,
			wkspID)
	}
	return deletedBindings, nil
}
```
Could you remove the option where transaction is not provided(keep one or the other) as we discussed? Also suggesting some add ons that will make it consistent with the code I wrote and be helpful for the WebUI code

Reply:
Changed and refactored tests!